### PR TITLE
DOC-730: Convert Storage reference to Markdown

### PR DESCRIPTION
**Pull request by taylorhixson**
Update the docs:

- Convert storage-reference.md to Markdown and review the functions.
- Edit structure.md to update Storage class overview
- Edit up42-reference.md to fix example

Corresponds to Documentation JIRA task [DOC-723](https://up42.atlassian.net/browse/DOC-723).

[DOC-723]: https://up42.atlassian.net/browse/DOC-723?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:38]:
```suggestion
| `created_after`    | **Union[str, datetime]**<br/>Search for assets created after the timestamp specified in the `YYYY-MM-DD` format.                                                                                                                                                                                                                                                                                                                         |
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:39]:
```suggestion
| `created_before`   | **Union[str, datetime]**<br/>Search for assets created before the timestamp specified in the `YYYY-MM-DD` format.                                                                                                                                                                                                                                                                                                                         |
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:40]:
```suggestion
| `workspace_id`     | **str**<br/>Search by workspace ID.                                                                                                                                                                                                                                                                                                                                                                                                     |
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:47]: For sortby, descending, and return_json, please reuse the text from https://sdk.up42.com/reference/project-reference/#get_jobs

Comment (dustgalactic) [reply docs/reference/storage-reference.md:47]: Sortby values will be different here + please remove `"`

Comment (dustgalactic) [reply docs/reference/storage-reference.md:47]: `</li>The default value is "createdAt". |`
You need to close the list with `</ul>`

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:71]: Let's also support [this release](https://github.com/up42/up42-py/pull/465/files#diff-f9cb3611528e0a7c83ebe23981dd51766dab5af53b9d6396e714552008d75f41R314) here:
- New arguments: 
  - `workspace_orders: bool = True`
  - `status: Optional[List[str]] = None`
  - `name: Optional[str] = None`
- Status values: 
  - ` "CREATED",
            "BEING_PLACED",
            "PLACED",
            "PLACEMENT_FAILED",
            "DELIVERY_INITIALIZATION_FAILED",
            "BEING_FULFILLED",
            "DOWNLOAD_FAILED",
            "DOWNLOADED",
            "FULFILLED",
            "FAILED_PERMANENTLY"` (+ a link to the [description](https://docs.up42.com/developers/api-tasking/tasking-monitor#order-statuses))

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:73]:
```suggestion
The `get_orders()` function allows you to search for tasking and catalog orders.
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:93]: Please make it consistent with the other `limit` description.

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:94]: The same as above

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:95]: The same as above

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:96]: Remove `"`

Comment (dustgalactic) [reply docs/reference/storage-reference.md:96]: "Don't use the parameter to get orders of all types." (I see that you took it from the reference, but I think there's a better way to describe it) -> To get orders of all types, omit the parameter.

Comment (dustgalactic) [reply docs/reference/storage-reference.md:96]: There's something wrong with line spacing above the sentence

File: docs/reference/storage-reference.md
Comment (dustgalactic) [reply docs/reference/storage-reference.md:44]: Actually, the Analytics team has just told me they dropped `ANALYTICS`

File: docs/reference/storage-reference.md
Comment (taylorhixson) [reply docs/reference/storage-reference.md:96]: After much investigation, I think the line spacing is like this because the sentence `To get orders of all types, omit the parameter.` doesn't have any text within the back ticks, like how the others have the default value written `True`. 
In my commit, I can add an empty ` ` at the end of the sentence, and it will add the line spacing. But moving this sentence to the top of the description or omitting the sentence entirely would solve the problem, too. 
What is your preference? @dustgalactic

Comment (dustgalactic) [reply docs/reference/storage-reference.md:96]: Let's move it to the top :)

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:97]:
```suggestion
| `limit`            | **int**<br/>The number of orders on a result page.                                                                                                                                                                                                                                                                                                                                                                  |
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:113]: Let's stick to the order in the table above: `order_type` first, then `status`.

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:95]:
```suggestion
| `workspace_orders` | **bool**<br/>Determines the scope of orders to return.<br/><ul><li>`True`: return orders from your workspace.</li><li>`False`: return orders from the entire account.</li></ul>The default value is `True`.                                                                                                                                                                                                                                        |
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:103]:
```suggestion
| `tags`             | **list[str]**<br/>Order tags. Use to search for orders with any of the provided tags.                                                                                                                                                                                                                                                                                                                                                  |
```

File: docs/reference/storage-reference.md
Comment (dustgalactic) [inline docs/reference/storage-reference.md:65]: Let's include `descending` in the example.

**Comment by dustgalactic** [comment]
@andher1802 @seedlit 
Changes introduced in this PR:
- Documentation-only:
  - Converted `storage-reference` to Markdown and updated `structure`.
  - Made small text updates in `project-reference`, `tasking-reference`, `up42-reference`, `webhook-reference`, `workflow-reference`.
- Code support:
  - In `get_assets()`, removed `USER` from `sources`. I received a confirmation this value should not be public. (@iliascholl-up42)
  - In `get_assets()`, replaced `ANALYTICS` from `PROCESSING` to support 1.0 -> 2.0 migration ([the related DocHub PR](https://github.com/up42/documentation-hub/pull/912/files) from @matheuspinheirosantos).
  - Supported the upcoming [v2 orders migration release](https://github.com/up42/up42-py/pull/465/files#diff-f9cb3611528e0a7c83ebe23981dd51766dab5af53b9d6396e714552008d75f41R314) by doing the adding new arguments to`get_orders()`: `workspace_orders`, `status`, `name`. So, this PR shouldn't be merged before the release.

**Review by dustgalactic — APPROVED** [review]

**Comment by seedlit** [comment]
@taylorhixson @dustgalactic 

> In get_assets(), replaced ANALYTICS from PROCESSING to support 1.0 -> 2.0 migration 

Thanks for spotting this.
Did you test that `ANALYTICS` is not working anymore?
And that `PROCESSING` is working?

**Comment by taylorhixson** [comment]
> Did you test that `ANALYTICS` is not working anymore? And that `PROCESSING` is working?

For me as of this morning, both `PROCESSING` and `ANALYTICS` are working. No results show up when I use `PROCESSING` but it doesn't throw an error either. @seedlit

**Comment by seedlit** [comment]
If you want to move forward, you can remove the v2 endpoint migration changes and merge this PR.
Or, we can wait for v2 endpoint PR to get merged first.
Your call.

**Comment by dustgalactic** [comment]
@seedlit Let's wait.

**Pull request merged**
